wrappers/wrappers/macs2/bdgpeakcall/wrapper.py

```diff
@@ -0,0 +1,80 @@
+import os
+import math
+import tempfile
+from snakemake import shell
+
+log = snakemake.log_fmt_shell()
+logfile = None
+extra = snakemake.params.get('extra', '')
+
+outdir, basebed = os.path.split(snakemake.output.bed)
+label = snakemake.params.block['label']
+extra = snakemake.params.block.get('extra', '')
+
+ip_bdg = snakemake.input.ip_bdg
+lambda_bdg = snakemake.input.lambda_bdg
+d_estimate = snakemake.input.d_estimate
+read_length = snakemake.input.read_length
+bed = snakemake.output.bed
+
+qvalue = snakemake.params.block.get('qvalue')
+if qvalue is None:
+    qvalue = 0.05
+else:
+    qvalue = float(qvalue[0])
+
+qvalue = -1 * math.log(qvalue, 10)
+
+if ip_bdg is None:
+    raise ValueError("macs2/bdgpeakcall requires input.ip_bdg")
+if lambda_bdg is None:
+    raise ValueError("macs2/bdgpeakcall requires input.lambda_bdg")
+if d_estimate is None:
+    raise ValueError("macs2/bdgpeakcall requires input.d_estimate")
+if read_length is None:
+    raise ValueError("macs2/bdgpeakcall requires input.read_length")
+if bed is None:
+    raise ValueError("macs2/bdgpeakcall requires output.bed")
+
+with open(d_estimate) as f:
+    d_estimate = f.readline().split()[2].strip()
+
+with open(read_length) as f:
+    read_length = f.readline().strip()
+
+tmp = tempfile.NamedTemporaryFile().name
+
+cmds = (
+    'macs2 bdgcmp '
+    '-t {ip_bdg} '
+    '-c {lambda_bdg} '
+    '-m qpois -o {tmp} '
+)
+shell(cmds + '{log}')
+
+cmds = (
+    'macs2 bdgpeakcall '
+    '-i {tmp} '
+    '-c {qvalue} '
+    '-l {d_estimate} '
+    '-g {read_length} '
+    '-o {bed} '
+)
+shell(cmds + '{log}')
+
+tmp2 = tempfile.NamedTemporaryFile().name
+# Fix the output file so that it doesn't have negative numbers and so it fits
```

Comment:
This fixing business ends up being a pretty common thing . . . should we move this to lcdblib so we can do `lcdblib.fix_bedlike(infile, chromsizes, outfile)`?

Reply:
I haven't actually observed the out of bounds problem in the wild. But yes it makes sense to move some sort of bed sanitizer to a more public location.